lib/renderer/api/desktop-capturer.js

```diff
@@ -1,46 +1,43 @@
-const ipcRenderer = require('electron').ipcRenderer
-const nativeImage = require('electron').nativeImage
+const {ipcRenderer, nativeImage} = require('electron')
 
-var nextId = 0
-var includes = [].includes
+const includes = [].includes
+let currentId = 0
 
-var getNextId = function () {
-  return ++nextId
+const incrementId = () => {
+  currentId += 1
+  return currentId
 }
 
 // |options.type| can not be empty and has to include 'window' or 'screen'.
-var isValid = function (options) {
-  return ((options != null ? options.types : void 0) != null) && Array.isArray(options.types)
+function isValid (options) {
+  const types = (options != null) ? options.types : undefined
+  return (types != null) && Array.isArray(options.types)
 }
 
 exports.getSources = function (options, callback) {
-  var captureScreen, captureWindow, id
-  if (!isValid(options)) {
-    return callback(new Error('Invalid options'))
-  }
-  captureWindow = includes.call(options.types, 'window')
-  captureScreen = includes.call(options.types, 'screen')
+  if (!isValid(options)) return callback(new Error('Invalid options'))
+  const captureWindow = includes.call(options.types, 'window')
+  const captureScreen = includes.call(options.types, 'screen')
+
   if (options.thumbnailSize == null) {
     options.thumbnailSize = {
       width: 150,
       height: 150
     }
   }
-  id = getNextId()
+
+  const id = incrementId()
   ipcRenderer.send('ELECTRON_BROWSER_DESKTOP_CAPTURER_GET_SOURCES', captureWindow, captureScreen, options.thumbnailSize, id)
-  return ipcRenderer.once('ELECTRON_RENDERER_DESKTOP_CAPTURER_RESULT_' + id, function (event, sources) {
-    var source
-    callback(null, (function () {
-      var i, len, results
-      results = []
-      for (i = 0, len = sources.length; i < len; i++) {
-        source = sources[i]
+  return ipcRenderer.once(`ELECTRON_RENDERER_DESKTOP_CAPTURER_RESULT_${id}`, (event, sources) => {
```

Comment:
Is it possible to shared some code between these two files?

Reply:
yes, but it's a small enough amount of shared code that i don't think the benefits outweigh the costs

Reply:
It's up to you ), sure.